pkg/cmd/release/download/download.go

```diff
@@ -245,6 +246,23 @@ func downloadAsset(httpClient *http.Client, assetURL, destinationPath string, ar
 		return api.HandleHTTPError(resp)
 	}
 
+	var destinationPath = filepath.Join(destinationDir, fileName)
+
+	if len(fileName) == 0 {
+		contentDisposition := resp.Header.Get("Content-Disposition")
+
+		_, params, err := mime.ParseMediaType(contentDisposition)
+		if err != nil {
+			return err
+		}
+		if serverFileName, ok := params["filename"]; ok {
+			destinationPath = filepath.Join(destinationDir, serverFileName)
+		} else {
+			return fmt.Errorf("unable to fetch file name from response's Content-Disposition header")
```

Comment:
Nitpick: this error message leaks too much about the implementation (the user doesn't care about a "Content-Disposition" header). Maybe something like:
```suggestion
			return errors.New("unable to determine file name of archive")
```